extensions/filter/README.md

```diff
@@ -0,0 +1,118 @@
+# Filter API Extension
+
+**Extension [Maturity Classification](../README.md#extension-maturity): Pilot**
+
+The STAC search endpoint, `/search`, by default only accepts the core filter parameters given in the *[api-spec](../../api-spec.md)*.
+The Filter API extension adds additional filters for searching on the properties of Items.
+
+This extension replaces the deprecated 'Query' extension, swapping the custom query language with OGC's [CQL](http://docs.opengeospatial.org/DRAFTS/19-079.html)
+([CQL repo](https://github.com/opengeospatial/ogcapi-features/tree/master/extensions/cql/)). It appears to still be a draft standard, 
+but using it now in STAC API will enable us to provide key feedback, and STAC API will only go to 1.0.0 when there is a stable target 
+to depend on. But CQL has been supported in a couple STAC Servers for awhile, and in the open source geospatial world for longer, so it 
+should be pretty stable.
+
+There are two flavors of CQL: `cql-text` and `cql-json`. STAC uses the former for HTTP GET requests, and the latter for HTTP POST requests. 
+The [CQL](http://docs.opengeospatial.org/DRAFTS/19-079.html) standard is the definitive reference for both flavors, and should work as described
+in the `/items` endpoints typical of an OGC API - Features implementation. STAC also uses the same constructs in the `/search` endpoint, in line
+with CQL's description of '[Cross-collection queries](http://docs.opengeospatial.org/DRAFTS/19-079.html#filter-param-multiple-collections)'.
+
+## Example
+
+Before we dive deep into an overview of CQL, it is worth showing a concrete examples. A common query of STAC is to find items (implementing the EO 
+extension) with cloud cover between 0 and 10%. This would look like: 
+
+### cql-text (GET)
+
+```http
+/search?filter=eo:cloud_cover < 10 
```

Comment:
With CQL you can query top-level fields, such as geometry with the intersects parameter. Do other filters all assume they are property names?
How would one query another top-level field like `assets` with CQL?
Or do normal properties have to be qualified, e.g., "properties.eo:cloud_cover"?

Reply:
I'm pretty sure CQL assumes the other filters all assume they are property names. With OGC I think it follows the 'feature model', which says a geometry and zero or more attributes - which in GeoJSON map to properties. But it's a good question - would be good for CQL to clarify its use with GeoJSON.

Reply:
to follow up on this, Part 3 requires that only queryables be used in expressions, so the implementer defines what the queryable-to-STAC field mapping is. 

I need to dig more into how (or if) one can do queries like "item has asset with band where common_band red", since `assets` is an array and `eo:bands` is an array.